src/Forms/FieldList.php

```diff
@@ -717,40 +683,29 @@ public function rootFieldList()
         return $this;
     }
 
-    /**
-     * @return CompositeField|null
-     */
-    public function getContainerField()
+    public function getContainerField(): ?CompositeField
     {
         return $this->containerField;
     }
 
-    /**
-     * @param CompositeField|null $field
-     * @return $this
-     */
-    public function setContainerField($field)
+    public function setContainerField(?CompositeField $field): static
     {
         $this->containerField = $field;
         return $this;
     }
 
     /**
      * Transforms this FieldList instance to readonly.
-     *
-     * @return FieldList
      */
-    public function makeReadonly()
+    public function makeReadonly(): static
```

Comment:
```suggestion
    public function makeReadonly(): FieldList
```
Subclasses might return a transformed list that ends up not being that subclass, but still ends up being a valid `FieldList`.

Reply:
Updated